File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        int ans=nums[0];
        int tempSum=0;
        for(int i=0;i<nums.length;i++){
            tempSum+=nums[i];
            ans=Math.max(ans,tempSum);
            if(tempSum<0)
                tempSum=0;
           
        }
        return ans;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& arr) {
        int currSum=0,ans=INT_MIN;
        for(int i=0;i<arr.size();i++){
            currSum+=arr[i];
            ans=max(currSum,ans);
            if(currSum<0)
                currSum=0;
            
        }
        return ans;
    }
};